exporter/signalfxexporter/correlation/correlation_test.go

```diff
@@ -33,20 +37,71 @@ func TestTrackerAddSpans(t *testing.T) {
 		},
 	)
 
+	err := tracker.Start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+	assert.NotNil(t, tracker.correlation, "correlation context should be set")
+
 	traces := pdata.NewTraces()
 	rs := traces.ResourceSpans().AppendEmpty()
 	attr := rs.Resource().Attributes()
 	attr.InsertString("host.name", "localhost")
 
 	// Add empty first, should ignore.
 	tracker.AddSpans(context.Background(), pdata.NewTraces())
-	assert.Nil(t, tracker.correlation)
 	assert.Nil(t, tracker.traceTracker)
 
 	tracker.AddSpans(context.Background(), traces)
 
-	assert.NotNil(t, tracker.correlation, "correlation context should be set")
 	assert.NotNil(t, tracker.traceTracker, "trace tracker should be set")
 
 	assert.NoError(t, tracker.Shutdown(context.Background()))
 }
+
+func TestTrackerStart(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		config  *Config
+		wantErr bool
+		errMsg  string
+	}{
+		{
+			name: "invalid http client settings fails",
+			config: &Config{
+				HTTPClientSettings: confighttp.HTTPClientSettings{
+					Endpoint: "localhost:9090",
+					TLSSetting: configtls.TLSClientSetting{
+						TLSSetting: configtls.TLSSetting{
+							CAFile: "/non/existent",
+						},
+					},
+				},
+			},
+			wantErr: true,
+			errMsg:  "failed to create correlation API client: failed to load TLS config",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tracker := NewTracker(
+				tt.config,
+				"abcd",
+				component.ExporterCreateParams{
+					Logger: zap.NewNop(),
+				},
+			)
+
+			err := tracker.Start(context.Background(), componenttest.NewNopHost())
+
+			if tt.wantErr {
+				require.Error(t, err)
+				if tt.errMsg != "" {
+					require.Contains(t, err.Error(), tt.errMsg)
```

Comment:
@dmitryax - I made a mistake with the past check. I switched it to a contains with a base portion of the error, so that the test works operating systems.